packages/frontend/src/pages/Dashboard.tsx

```diff
@@ -273,6 +275,40 @@ const Dashboard: FC = () => {
         setHaveTilesChanged,
     ]);
 
+    const handleToggleFullscreen = () => {
+        const willBeFullscreen = !isFullscreen;
+
+        if (document.fullscreenElement && !willBeFullscreen) {
+            document.exitFullscreen();
+        } else if (
+            document.fullscreenEnabled &&
+            !document.fullscreenElement &&
+            willBeFullscreen
+        ) {
+            document.documentElement.requestFullscreen();
+        }
+
+        toggleFullscreen();
+    };
+
+    useEffect(() => {
+        const onFullscreenChange = () => {
+            if (isFullscreen && !document.fullscreenElement) {
+                toggleFullscreen(false);
+            } else if (!isFullscreen && document.fullscreenElement) {
+                toggleFullscreen(true);
+            }
+        };
+
+        document.addEventListener('fullscreenchange', onFullscreenChange);
+
+        return () =>
+            document.removeEventListener(
+                'fullscreenchange',
+                onFullscreenChange,
+            );
+    });
```

Comment:
this is great, but we can also use this v useful hook 😉 https://mantine.dev/hooks/use-fullscreen/

Reply:
Awesome, thanks for the heads up -- I keep getting surprised by features built into mantine 😄 I'll refactor with this hook.